Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Slate/HandInteractionPanZoom.cs

```diff
@@ -260,79 +285,69 @@ private void UpdateIdle()
+            float scaleUVDelta = 0.0f;
+            Vector2 scaleUVCentroid = Vector2.zero;
+            float currentContactRatio = 0.0f;
+
+            if (scaleActive)
             {
-                bool oobX = false;
-                bool oobY = false;
-                Vector2 tiling = new Vector2(1.0f, 1.0f);
-                List<Vector2> uvs = new List<Vector2>();
-                List<Vector2> uvsOrig = new List<Vector2>();
-                mesh.GetUVs(0, uvs);
-                uvsOrig.AddRange(uvs);
-                float scaleUVDelta = 0.0f;
-                Vector2 scaleUVCentroid = Vector2.zero;
-                float currentContactRatio = 0.0f;
+                scaleUVCentroid = GetDisplayedUVCentroid();
+                currentContactRatio = GetUVScaleFromTouches();
+                scaleUVDelta = currentContactRatio / previousContactRatio;
+                previousContactRatio = currentContactRatio;
 
-                if (scaleActive)
+                for (int i = 0; i < uvs.Count; ++i)
                 {
-                    //scaleUVCentroid = GetScaleUVCentroid();
-                    scaleUVCentroid = GetDisplayedUVCentroid();
-                    currentContactRatio = GetUVScaleFromTouches();
-
-                    //todo fix this- totalUVScale resets to 1 when second hand touches.
-
-                    //hint: totalUVOffset works correctly- make scaleUVDelta work the same way
-                    scaleUVDelta = currentContactRatio / previousContactRatio;
-                    previousContactRatio = currentContactRatio;
+                    //this is where zoom is applied if Active
+                    uvs[i] = ((uvs[i] - scaleUVCentroid) / scaleUVDelta) + scaleUVCentroid;
                 }
+            }
 
+            //test for pan limits
+            Vector2 uvDelta = new Vector2(totalUVOffset.x, -totalUVOffset.y);
+            if (unlimitedPan == false)
+            {
+                bool xLimited = false;
+                bool yLimited = false;
                 for (int i = 0; i < uvs.Count; ++i)
                 {
-                    if (scaleActive)
+                    uvTestValue = uvs[i] - uvDelta;
+                    if (uvTestValue.x > tiling.x * maxPanHorizontal || uvTestValue.x < -(tiling.x * maxPanHorizontal))
                     {
-                        //this is where zoom is applied if Active
-                        uvs[i] = ((uvs[i] - scaleUVCentroid) / scaleUVDelta) + scaleUVCentroid;
+                        xLimited = true;
                     }
-
-
-                    //this is where the Pan is applied
-                    uvs[i] = new Vector2(uvs[i].x - (totalUVOffset.x / previousContactRatio), uvs[i].y + (totalUVOffset.y / previousContactRatio));
-
-
-                    //Pan limits are applied here if specified
-                    if (wrapTexture == false)
+                    if (uvTestValue.y > tiling.y * maxPanVertical || uvTestValue.y < -(tiling.y * maxPanVertical))
```

Comment:
Not actually a real suggestion, but wanted to bring it up for fun - since this is just a magnitude test this could have also been like testXValue > Math.Abs(tiling.x * maxPanHorizontal). Either or is fine :)

Reply:
Feel free to ignore this and close this :)